src/border_agent/border_agent.cpp

```diff
@@ -579,51 +506,28 @@ void BorderAgent::PublishMeshCopService(void)
     {
         txtList.emplace_back("mn", mProductName.c_str());
     }
-    txtList.emplace_back("nn", networkName);
-    txtList.emplace_back("xp", extPanId->m8, sizeof(extPanId->m8));
-    txtList.emplace_back("tv", mHost.GetThreadVersion());
-
-    // "xa" stands for Extended MAC Address (64-bit) of the Thread Interface of the Border Agent.
-    txtList.emplace_back("xa", extAddr->m8, sizeof(extAddr->m8));
-    state                 = GetStateBitmap(*instance);
-    state.mEpskcSupported = GetEphemeralKeyEnabled();
-    stateUint32           = htobe32(state.ToUint32());
-    txtList.emplace_back("sb", reinterpret_cast<uint8_t *>(&stateUint32), sizeof(stateUint32));
-
-    if (state.mThreadIfStatus == kThreadIfStatusActive)
-    {
-        uint32_t partitionId;
 
-        AppendActiveTimestampTxtEntry(*instance, txtList);
-        partitionId = otThreadGetPartitionId(instance);
-        txtList.emplace_back("pt", reinterpret_cast<uint8_t *>(&partitionId), sizeof(partitionId));
-    }
+    AppendVendorTxtEntries(mMeshCoPTxtUpdate, txtList);
 
-#if OTBR_ENABLE_BACKBONE_ROUTER
-    AppendBbrTxtEntries(*instance, state, txtList);
-#endif
-#if OTBR_ENABLE_BORDER_ROUTING
-    AppendOmrTxtEntry(*instance, txtList);
-#endif
+    // When thread interface is not active, the border agent is not started, thus it's not listening to any port and
+    // not handling requests. In such situation, we use a dummy port number for publishing the MeshCoP service to
+    // advertise the status of the border router. One can learn the thread interface status from `sb` entry so it
+    // doesn't have to send requests to the dummy port when border agent is not running.
+    port = mBaIsActive ? mMeshCoPUdpPort : kBorderAgentServiceDummyPort;
 
-    AppendVendorTxtEntries(mMeshCopTxtUpdate, txtList);
+    error = Mdns::Publisher::EncodeTxtData(txtList, txtData);
+    assert(error == OTBR_ERROR_NONE);
 
-    if (otBorderAgentIsActive(instance))
+    // If `txtList` is empty, `txtData` will have one byte 0x00 as implemented in ` Mdns::Publisher::EncodeTxtData`.
```

Comment:
This sounds hacky. Is there a more elegant way to handle this?

Reply:
Updated `Mdns::Publisher::EncodeTxtData` and remove the check here.

Reply:
Updated to a more elegant way: only call `EncodeTxtData` when `txtList` is not empty. Then we will absolutely get a non-empty `txtData`.